mesa/datacollection.py

```diff
@@ -180,11 +178,14 @@ def _record_agents(self, model):
         rep_funcs = self.agent_reporters.values()
 
         def get_reports(agent):
-            _prefix = (agent.model.schedule.steps, agent.unique_id)
+            _prefix = (agent.model._steps, agent.unique_id)
             reports = tuple(rep(agent) for rep in rep_funcs)
             return _prefix + reports
 
-        agent_records = map(get_reports, model.schedule.agents)
+        agent_records = map(
+            get_reports,
+            model.schedule.agents if hasattr(model, "schedule") else model.agents,
+        )
```

Comment:
Really nice! Could you add tests for this behaviour? Just to be 100% certain it works as expected.

Reply:
This is such a simple line (far simpler than AgentSet's shuffle) and is overkill to add such. That said, there was and is a test:
1. you have seen the CI tests passing with either of the if case.
2. there is already the experimental Schelling in place, which has no scheduler.

Reply:
What is missing is test of multiple schedulers and datacollectors pairs

Reply:
Yes, correcting for multiple scheduler-datacollector pairs is out of scope for this PR.